extension/chrome/elements/compose-modules/formatters/general-mail-formatter.ts

```diff
@@ -80,4 +89,19 @@ export class GeneralMailFormatter {
     // throws ComposerResetBtnTrigger when user closes pass phrase dialog without entering
     return await view.storageModule.decryptSenderKey(parsedSenderPrv);
   };
+
+  private static isSenderInKeyUserIds = async (view: ComposeView, signingKey?: ParsedKeyInfo | undefined): Promise<boolean> => {
+    const senderEmail = view.senderModule.getSender();
+    const storage = await AcctStore.get(view.acctEmail, ['sendAs']);
+    const emailAliases = Object.keys(storage.sendAs || {});
+    if (senderEmail !== view.acctEmail && signingKey?.keyInfo.emails) {
+      const senderAliasIndex = emailAliases.indexOf(view.acctEmail);
+      if (senderAliasIndex !== -1) {
+        emailAliases.splice(senderAliasIndex, 1);
+      }
+      const emailsInSigningKey = signingKey.keyInfo.emails;
+      return emailsInSigningKey.some(email => emailAliases.includes(email));
```

Comment:
hmm, not sure if it's correct:

for example, we have user with `view.acctEmail = test.user@gmail.com`, it has several aliases - `test.user@gmail.com`, `testuser@gmail.com`, `test+user@gmail.com`.

and signing key has `signingKey.keyInfo.emails = ['test.user@gmail.com', 'testuser@gmail.com']`

then user tries to send message using alias `test+user@gmail.com` and there won't be any error as `emailsInSigningKey` will include `testuser@gmail.com`

or I'm missing something?

Reply:
Ah yes, thanks for noticing. I thought those filters in email only applies in inboxes... There wouldn't be any error indeed in that case you have mentioned. I'll update it too.

Reply:
updated. thanks!